src/java.base/share/classes/java/lang/invoke/BootstrapMethodInvoker.java

```diff
@@ -243,65 +254,54 @@ private static Object invokeWithManyArguments(MethodHandle bootstrapMethod, Look
-            Lookup.class, String.class, Class.class, MethodType.class, MethodHandle.class, MethodType.class);
-
-    private static final MethodType SCF_MT = MethodType.methodType(CallSite.class,
-            Lookup.class, String.class, MethodType.class, String.class, Object[].class);
-
     /**
-     * @return true iff the BSM method type exactly matches
-     *         {@link java.lang.invoke.StringConcatFactory#makeConcatWithConstants(MethodHandles.Lookup,
-     *                 String,MethodType,String,Object...))}
+     * Exact type used for the {@link java.lang.runtime.ObjectMethods#bootstrap(
+     * MethodHandles.Lookup,String,TypeDescriptor,Class,String,MethodHandle[])} bootstrap method.
      */
-    private static boolean isStringConcatFactoryBSM(MethodType bsmType) {
-        return bsmType == SCF_MT;
-    }
+    private static final MethodType OBJECT_METHODS_MT = MethodType.methodType(Object.class,
+            Lookup.class, String.class, TypeDescriptor.class, Class.class, String.class, MethodHandle[].class);
 
     /**
-     * @return true iff the BSM method type exactly matches
-     *         {@link java.lang.invoke.LambdaMetafactory#metafactory(
-     *          MethodHandles.Lookup,String,Class,MethodType,MethodHandle,MethodType)}
+     * Exact type of the bootstrap methods generated for {@link java.lang.reflect.Proxy} classes.
      */
-    private static boolean isLambdaMetafactoryCondyBSM(MethodType bsmType) {
-        return bsmType == LMF_CONDY_MT;
-    }
+    private static final MethodType PROXY_GET_METHOD_MT = MethodType.methodType(Method.class,
+            Lookup.class, String.class, Class.class, Class.class, String.class, MethodType.class);
 
     /**
-     * @return true iff the BSM method type exactly matches
-     *         {@link java.lang.invoke.LambdaMetafactory#metafactory(
-     *          MethodHandles.Lookup,String,MethodType,MethodType,MethodHandle,MethodType)}
+     * Exact type used of the {@link java.lang.invoke.ConstantBootstraps#getStaticFinal(Lookup, String, Class)}
+     * bootstrap method.
      */
-    private static boolean isLambdaMetafactoryIndyBSM(MethodType bsmType) {
-        return bsmType == LMF_INDY_MT;
-    }
+    private static final MethodType CONDY_GET_STATIC_FINAL_MT = MethodType.methodType(Object.class,
+            Lookup.class, String.class, Class.class);
 
     /**
-     * @return true iff the BSM method type exactly matches
-     *         {@link java.lang.invoke.LambdaMetafactory#altMetafactory(
-     *          MethodHandles.Lookup,String,MethodType,Object[])}
+     * Exact type of the {@link java.lang.invoke.ConstantBootstraps#getStaticFinal(Lookup, String, Class, Class)}
+     * bootstrap method.
      */
-    private static boolean isLambdaMetafactoryAltMetafactoryBSM(MethodType bsmType) {
-        return bsmType == LMF_ALT_MT;
-    }
+    private static final MethodType CONDY_GET_STATIC_FINAL_CLASS_MT = MethodType.methodType(Object.class,
```

Comment:
I recommend calling this `CONDY_EXTRA_CLASS`, as this is the type of `arrayVarHandle`.